internal/pkg/api/plugin.go

```diff
@@ -0,0 +1,94 @@
+//      | | (Parser)  |     | Mem/File | |   |  -----------                 -----------  |
+//      |  -----------       ----------  |   |      ||                          ||       |
+//       --------------------------------    |      \/	                        \/       |
+//                                           |  ---------------------------------------  |
+//                                           | |             OutputEventContext        | |
+//                                           |  ---------------------------------------  |
+//                                            -------------------------------------------
+//                             				   ||
+//                                             ||        --------------------------------
+//                                             ||     ->|         Sharing Client        |
+//                                             ||    |   --------------------------------
+//                                             ||    |
+//                                             \/    |            SegmentSender
+//                                            ---    |  ---------------------------------
+//                                           |   |   | |  -----------       -----------  |
+//                                           | D |   |-| |BatchBuffer| ==> | Forwarder | |
+//                                           | i |   | |  -----------       -----------  |
+//                                           | s |   |  ---------------------------------
+//                                           | p |   |
+//                                           | a |=> |              .......                 ===> Kafka/OAP
+//                                           | t |   |
+//                                           | c |   |             MeterSender
+//                                           | h |   | -----------------------------------
+//                                           | e |   -|  -------------       -----------  |
+//                                           | r |    | | BatchBuffer | ==> | Forwarder | |
+//                                           |   |    |  -------------       -----------  |
+//                                            ---      -----------------------------------
+//
+//
+// 1. The Collector plugin would fetch or receive the input data.
+// 2. The Parser plugin would parse the input data to SerializationEvent that is supported
+//    to be stored in Queue.
+// 3. The Queue plugin stores the SerializationEvent. However, the serialization depends on
+//    the Queue implements. For example, the serialization is unnecessary when using a Memory
+//    Queue.
+// 4. The Filter plugin would pull an event from the Queue and process the event to create
+//    a new event. Next, the event is passed to the next filter to do the same things until
+//    the whole filters are performed. The events labeled with RemoteEvent type would be
```

Comment:
I think we should add an `output` catalog in the event directly, right? And every type as `output` should have a collection of forwarder(s), right?

Reply:
Do u means the following things? 
![image](https://user-images.githubusercontent.com/31562192/100545366-77b89200-3296-11eb-8c76-eb60dd3a20bf.png)

Reply:
add comments in plugin.go